src/App.jsx

```diff
@@ -1,35 +1,89 @@
+import React, { useCallback, useEffect, useState } from "react";
 import Body from "./Layouts/body/Body";
 import Footer from "./Layouts/footer/Footer";
 import Header from "./Layouts/header/Header";
 import ErrorBoundary from "./Components/ErrorBoundary";
-import { api } from './API/PostService'
+import { api } from './API/PostService';
 
-import VidoeModal from './Components/UI/videomodal/VideoModal'
-import VideoForm from './Components/UI/videoform/VideoForm'
+import Modal from './Components/UI/modal/Modal';
+import VideoFormModal from './Components/UI/videoformmodal/VideoFormModal';
+import VideoDetails from "./Components/VideoDetails";
+import MyButton from './Components/UI/button/MyButton';
+import useModalState from './hooks/useModalState';
 
 const App = () => {
 
   const [videos, setVideos] = useState([]);
-  const [showModal, setShowModal] = useState(false);
+  const [isVideoDetail, setIsVideoDetail] = useState(false);
+  const [videoDetaill, setVideoDetaill] = useState();
+
+  const modal = useModalState();
 
   useEffect(() => {
     setVideos(api);
   }, []);
 
   const createVideo = (newVideo) => {
+    newVideo.id = videos.length+1;
+
     setVideos([...videos, newVideo]);
+  };
+
+  const videoDetail = (id) => {
+    console.log("render");
+    setVideoDetaill(videos[id - 1]);
+    setIsVideoDetail(true);
+  };
+
+  const editVideo = (video) => {
+    const editedVideos = videos.map(current => {
+      return current.id === video.id ? video : current;
+    });
+
+    setVideos(editedVideos);
   }
 
+  const removeVideo = (video) => {
+    setVideos(videos.filter(v => v.id !== video.id));
+  };
+
   return(
     <>
       <ErrorBoundary>
-        <VidoeModal className='videoModal' title='add movie' showModal={showModal} setShowModal={setShowModal}>
-          <VideoForm creat={createVideo} showModal={setShowModal} />
-        </VidoeModal>
+        {modal.isAddType &&
+         <Modal className='videoModal' title='add movie' onModalClose={modal.close}>
+           <VideoFormModal onSubmit={createVideo} hideModal={modal.close} />
+         </Modal>
+        }
+        {modal.isEditType &&
+          <Modal className='videoModal' title='edit movie' onModalClose={modal.close}>
+            <VideoFormModal editVideo={modal.data} onSubmit={editVideo} hideModal={modal.close} />
+          </Modal>
+        }
+        {modal.isConfirmType &&
+          <Modal
+            className='videoModal'
+            title='delete movie'
+            text='Are you sure you want to delete this movie?'
+            onModalClose={modal.close}
+          >
+            <MyButton className='button__red' onClick={() => {removeVideo(modal.data); modal.close()}}>confirm</MyButton>
+          </Modal>
+        }
 
-        <Header showModal={setShowModal}/>
-        <Body videos={videos} setVideos={setVideos} />
+        {isVideoDetail
+          ?
+            <VideoDetails video={videoDetaill} setIsVideoDetail={setIsVideoDetail} />
```

Comment:
it looks like we are using this callback only to reset video detail... i would recommend not to pass setter, but reset function instead... and also call this property smth like onVideoDetailsClose